scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py

```diff
@@ -237,3 +234,26 @@ def update_current_groups_list(self):
     def update_current_phase_tables(self):
         phase_table_list = self.context.phase_context.get_phase_table_list(self.context.data_context.instrument)
         self.view.set_phase_table_combo_box(phase_table_list)
+
+    def handle_first_good_data_changed(self):
+        self._validate_data_changed(self.view.first_good_time, "First Good Data")
+
+    def handle_last_good_data_changed(self):
+        self._validate_data_changed(self.view.last_good_time, "Last Good Data")
+
+    def _validate_data_changed(self, data, string):
+        run = float(get_run_number_from_workspace_name(self.view.input_workspace,
+                                                       self.context.data_context.instrument))
+        last_good_time = self.context.last_good_data([run])
+        first_good_time = self.context.first_good_data([run])
+
+        if self.view.first_good_time > self.view.last_good_time:
+            self.view.first_good_time = first_good_time
+            self.view.last_good_time = last_good_time
+            self.view.warning_popup("First Good Data cannot be greater than Last Good Data")
```

Comment:
Ah this is my mistake. I forgot to mention this branch is based from a different branch to master, and so these changes are in fact from #30257, but to answer your question: There are 3 possible cases. One case the user has changed first or last and it has results in first being greater than last, this is represented by the first if and uses first and last from the view. The other two cases are comparing the users changed data to the minimum and maximum values that they can be. since there was overlap between the code for first and last I merged them together which is why a particular case might seem redundant, for example first being greater than the max, as the first check will produce a warning that first is greater than last before being able to check it is greater than the max, and vice versa. I hope that made some sort of sense.